app.py

```diff
@@ -595,6 +608,126 @@ def call_api(url, params=None):
+            team_roster[full_name] = stats_cleanup(player['playerPoolEntry']['player']['stats'])
+        # Store the roster and each player's stats in a dict of team roster by team names
+        team_rosters[team_dict[team["id"]]] = team_roster
+    return team_rosters
+
+
+def stats_cleanup(stats):
+    # The stats dict has a couple of different average stats. We can find the actual stat timeframe using the
+    # following info:
+    # seasonId: season end year
+    # stat source: 0 = actual, 1 = projected
+    # stat split type: 0 = season, 1 = 7 days, 2 = 15 days, 3 = 30 days
+    current_year = int(app.config.get("SEASON"))
+    previous_year = current_year - 1
+    cleaned_up_stats = {}
+    for stat in stats:
+        if stat['seasonId'] == current_year and stat['statSourceId'] == 0:
+            if stat['statSplitTypeId'] == 0:
+                cleaned_up_stats[str(current_year) + " Season"] = {"FG%": 0, "FGM": 0, "FGA": 0, "FT%": 0, "FTM": 0,
+                                                                   "FTA": 0, "3PM": 0,
+                                                                   "REB": 0, "AST": 0, "STL": 0, "BLK": 0, "TO": 0,
+                                                                   "PTS": 0, "GP": 0}
+            elif stat['statSplitTypeId'] == 1:
+                cleaned_up_stats["Last 7"] = {"FG%": 0, "FGM": 0, "FGA": 0, "FT%": 0, "FTM": 0, "FTA": 0, "3PM": 0,
+                                              "REB": 0, "AST": 0, "STL": 0, "BLK": 0, "TO": 0, "PTS": 0}
+            elif stat['statSplitTypeId'] == 2:
+                cleaned_up_stats["Last 15"] = {"FG%": 0, "FGM": 0, "FGA": 0, "FT%": 0, "FTM": 0, "FTA": 0, "3PM": 0,
+                                               "REB": 0, "AST": 0, "STL": 0, "BLK": 0, "TO": 0, "PTS": 0}
+            elif stat['statSplitTypeId'] == 3:
+                cleaned_up_stats["Last 30"] = {"FG%": 0, "FGM": 0, "FGA": 0, "FT%": 0, "FTM": 0, "FTA": 0, "3PM": 0,
+                                               "REB": 0, "AST": 0, "STL": 0, "BLK": 0, "TO": 0, "PTS": 0}
+        elif stat['seasonId'] == current_year:
+            cleaned_up_stats[str(current_year) + " Projections"] = {"FG%": 0, "FGM": 0, "FGA": 0, "FT%": 0, "FTM": 0,
+                                                                    "FTA": 0,
+                                                                    "3PM": 0, "REB": 0, "AST": 0, "STL": 0, "BLK": 0,
+                                                                    "TO": 0, "PTS": 0}
+        elif stat['seasonId'] == previous_year and stat['statSourceId'] == 0:
+            cleaned_up_stats[str(previous_year) + " Season"] = {"FG%": 0, "FGM": 0, "FGA": 0, "FT%": 0, "FTM": 0,
+                                                                "FTA": 0, "3PM": 0,
+                                                                "REB": 0, "AST": 0, "STL": 0, "BLK": 0, "TO": 0,
+                                                                "PTS": 0}
+
```

Comment:
I'm not sure what's going on here. It looks like all we're doing is ensuring that `cleaned_up_stats` is set to a certain default. If that's the case, I think there are simpler ways of expressing that rather than multiple nested if-else statements. I'm happy to refactor this if you'd like.

My bigger concern is that we're hardcoding categories. Wherever possible, I've tried to ensure that the tools on this website work for leagues that use any combination of categories in their scoring. What happens if the league is using non-default categories in their scoring?

Can you think of a way to address this?